numpy/lib/function_base.py

```diff
@@ -764,6 +764,22 @@ def copy(a, order='K', subok=False):
     >>> x[0] == z[0]
     False
 
+    Note that ``np.copy`` is a shallow copy and will not
+    copy object elements within arrays:
+
+    >>> obj = dict()
+    >>> a = np.array([obj], dtype=object)
+    >>> b = np.copy(a)
+    >>> b[0] is a[0]
+    True
```

Comment:
```suggestion
    >>> a = np.array([1, 'm', [2, 3, 4]], dtype=object)
    >>> b = np.copy(a)
    >>> b[2][0] = 10
    >>> a
    array([1, 'm', list([10, 3, 4])], dtype=object)
```
I'm still not sure whether an example is necessary as I think Python programmers will generally already be familiar with the distinction between shallow/deep copies. Nevertheless, if we do decide to include examples, maybe an `object` array with more familiar elements could demonstrate the point more clearly?

Reply:
Thanks for the tip @rossbar. This was my original approach, although it wasn't this clear.